### PR TITLE
Add some more AssertThrowMPIs.

### PR DESCRIPTION
**Pull request by drwells**

### CODE REVIEW

**Review by tjhei — APPROVED** [review]

File: include/deal.II/base/aligned_vector.h [hunk 1]
```diff
@@ -1536,7 +1536,8 @@ AlignedVector<T>::replicate_across_communicator(const MPI_Comm &   communicator,
 
   // Make sure that the shared memory host has copied the data before we try to
   // access it.
-  MPI_Barrier(shmem_group_communicator);
+  const int ierr = MPI_Barrier(shmem_group_communicator);
+  AssertThrowMPI(ierr);
```
Comment (bangerth) [inline include/deal.II/base/aligned_vector.h:1540]: Ha, funny -- I found that same place yesterday!

**Comment by bangerth** [comment]
Nice! How did you actually find all of these places?

**Comment by drwells** [comment]
Thanks. I searched for `^ *MPI_.*(` - there were a few false positives but that regular expression matched most of our unchecked MPI calls.

**Comment by bangerth** [comment]
Ah, I thought that you had then also done `grep -A2 ...` to see whether the line following MPI calls has an `AssertThrowMPI`. But then you would have found #12321. Maybe I should do that at some point....

**Comment by bangerth** [comment]
Ah, that's quite complicated because so many of the MPI calls extend over several lines :-(

**Pull request merged**